lms/models/application_instance.py

```diff
@@ -30,6 +30,12 @@ class ApplicationInstance(BASE):  # pylint:disable=too-few-public-methods
         server_default=sa.sql.expression.true(),
         nullable=False,
     )
+    canvas_sections_enabled = sa.Column(
+        sa.Boolean(),
+        default=False,
+        server_default=sa.sql.expression.false(),
+        nullable=False,
+    )
```

Comment:
Exactly the same as the `provisioning` column above

Reply:
I don't want to get in to refactoring `ApplicationInstance` now, but some notes:

There's no such thing as a "Canvas application instance" or a "Moodle application instance" etc. There are just application instances, and one of the things that an application instance can have is a `canvas_sections_enabled` flag. They also have a `canvas_developer_key` and `canvas_developer_secret` but the `canvas_` prefix is missing from those Canvas-specific fields.

An application instance is just a thing that gets created when you fill in the https://lms.hypothes.is/welcome form. At the time when you submit the form we don't know whether you're going to use it in Canvas, Moodle, or something else. You don't even have to use it in just one: you could use it first in Canvas and then install it in Moodle.

What an application instance really is, is an LTI 1 consumer key and shared secret. You enter some info into `/welcome`--a domain name (that isn't really used for anything), email address (just for our records), and optionally a Canvas developer ID and key--and we generate a consumer key and shared secret for you and associate the bits of into you submitted with that consumer key and shared secret.

A correct domain model for this might be something like:

1. A `consumer_key` table (with `key` and `secret` columns)
2. Some registration info (domain and email) possibly in a separate table with a foreign key to the `consumer_key` table or possibly just in extra columns on the `consumer_key` table
3. A Canvas developer key (with `id` and `key` columns), possibly in a separate `canvas_developer_key` table with a foreign key to the `consumer_key` table or possibly just in extra columns on the `consumer_key` table

It might be worth a look through what the LTI specs and guides call this though, for naming and modeling ideas.

But the consumer keys (a.k.a. "application instances") themselves aren't LMS-specific. There aren't different types of application instance, like Canvas app instance and Moodle app instance etc.